packages/core/src/components/badge/badge.tsx

```diff
@@ -15,47 +15,42 @@ export class TdsBadge {
   /** Sets component size. */
   @Prop() size: 'lg' | 'sm' = 'lg';
 
-  @State() shape: string = '';
-
-  @State() text: string = '';
-
-  @Watch('value')
-  @Watch('size')
-  watchProps() {
-    this.checkProps();
-  }
-
-  componentWillLoad() {
-    this.checkProps();
-  }
-
-  checkProps() {
+  private computeBadgeData() {
     const valueAsNumber = parseInt(this.value);
+    let shape = '';
+    let text = '';
+    let ariaLabel = '';
+
     if (!Number.isNaN(valueAsNumber) && this.size !== 'sm') {
-      this.shape = this.value.toString().length >= 2 ? 'pill' : '';
-      this.size = 'lg';
-      this.text = valueAsNumber.toString().length >= 3 ? '99+' : valueAsNumber.toString();
+      shape = this.value.length >= 2 ? 'pill' : '';
+      text = valueAsNumber.toString().length >= 3 ? '99+' : valueAsNumber.toString();
+      ariaLabel = `Notification badge with ${text} new notifications`;
```

Comment:
Should this not be set by the users? Might want to have a text in some other language?

Reply:
Good point, @MartinPiq. Since the repository currently only supports English, I wanted to maintain consistency by following the same approach here. Given that the component already has access to the value, I thought it made sense to handle the logic for the aria-label within the component itself.

Reply:
Since you also reacted to Martins comment, please give your thoughts as input @timrombergjakobsson